src/h3/_cy/geo.pyx

```diff
@@ -0,0 +1,270 @@
+cimport h3lib
+from h3lib cimport bool, H3int
+from .util cimport (
+    check_cell,
+    check_edge,
+    check_res,
+    create_ptr,
+    create_mv,
+)
+from libc cimport stdlib
+
+
+cdef h3lib.GeoCoord deg2coord(double lat, double lng):
+    cdef:
+        h3lib.GeoCoord c
+
+    c.lat = h3lib.degsToRads(lat)
+    c.lng = h3lib.degsToRads(lng)
+
+    return c
+
+
+cdef (double, double) coord2deg(h3lib.GeoCoord c):
+    return (
+        h3lib.radsToDegs(c.lat),
+        h3lib.radsToDegs(c.lng)
+    )
+
+
+cpdef H3int geo_to_h3(double lat, double lng, int res) except 1:
+    cdef:
+        h3lib.GeoCoord c
+
+    check_res(res)
+
+    c = deg2coord(lat, lng)
+
+    return h3lib.geoToH3(&c, res)
+
+
+cpdef (double, double) h3_to_geo(H3int h) except *:
+    """Map an H3 cell into its centroid geo-coordinate (lat/lng)"""
+    cdef:
+        h3lib.GeoCoord c
+
+    check_cell(h)
+
+    h3lib.h3ToGeo(h, &c)
+
+    return coord2deg(c)
+
+
+cdef h3lib.Geofence make_geofence(geos, bool lnglat_order=False) except *:
+    """
+
+    Parameters
+    ----------
+    geos : list or tuple
+        GeoFence: A sequence of >= 3 (lat, lng) pairs where the last
+        element may or may not be same as the first (to form a closed loop).
+        The order of the pairs may be either clockwise or counterclockwise.
+    lnglat_order : bool
+        If True, assume coordinate pairs like (lng, lat)
+        If False, assume coordinate pairs like (lat, lng)
+    """
+    cdef:
+        h3lib.Geofence gf
+
+    gf.numVerts = len(geos)
+
+    gf.verts = <h3lib.GeoCoord*> stdlib.calloc(gf.numVerts, sizeof(h3lib.GeoCoord))
```

Comment:
May want to note in this function's documentation the need for freeing.

Reply:
Good point. Added.